go/vt/tabletserver/sqlquery.go

```diff
@@ -175,12 +190,12 @@ func (sq *SqlQuery) disallowQueries() {
 	sq.mu.Unlock()
 	// Terminate all streaming queries
 	sq.qe.streamQList.TerminateAll()
-	// Don't hold lock while waiting.
+	// Wait for outstanding requests to finish.
 	sq.requests.Wait()
 
-	// SHUTTING_QUERIES -> NOT_SERVING
-	sq.mu.Lock()
 	defer func() {
```

Comment:
I agree on removing panic as error mechanism. We have been talking about this a while, we should do it in the background, one step at a time.

However, we still need to catch panics for each rpc / high level API entry point, to protect the server. It's safer than crashing the process, as long as we work on them too...

In this case, however, this is a tough call: if qe.Close() panics, we have no idea in which state it is. Restarting qe when we get healthy is probably not gonna work. So we have to think carefully about all this. Maybe protecting RPCs from panics is the way to go only, and not the management stuff like this...

Not for this change though.